reactive/kotlinx-coroutines-reactor/src/Mono.kt

```diff
@@ -68,11 +68,13 @@ private class MonoCoroutine<in T>(
 
     override fun onCancelled(cause: Throwable, handled: Boolean) {
         try {
-            /*
-             * sink.error handles exceptions on its own and, by default, handling of undeliverable exceptions is a no-op.
-             * Guard potentially non-empty handlers against meaningless cancellation exceptions
-             */
-            if (getCancellationException() !== cause) {
+            if (getCancellationException() === cause) {
```

Comment:
Both current documentation and test seem to be really suspicious.

I think I got the idea why the code was written this way (also, `blame` was kinda useful): we, in fact, have _two_ cancellation modes: 
1) The coroutines was cancelled because the subscriber cancelled the subscription.
2) The coroutine itself has been cancelled due to timeouts or any other unhandled cancellation exceptions.

I've asked @sdeleuze about this, but current behaviour seems to be error-prone.
I'd propose the following instead:

1) If the subscriber cancelled the coroutine, then such cancellation is "expected" and CE should not be reported anywhere
2) If the coroutine was cancelled on its own, then it's a programmatic error that should be delivered.

Also, that way it will be consistent with `rxSingle` (its worth checking its implementation as well)